### ISSUE TITLE
Bad error report for malformed OSGi headers

### ISSUE THREAD

**Issue by laeubi** [eclipse-tycho/tycho#328]
Given one uses the following in a target:

```
<location includeSource="true" missingManifest="generate" type="Maven">
	<groupId>javax.xml.ws</groupId>
	<artifactId>jaxws-api</artifactId>
	<version>2.3.1</version>
	<type>jar</type>
</location>
```

results in the following error:

> [ERROR] Failed to resolve target definition Artifact MavenArtifactFacade [wrappedArtifact=javax.xml.ws:jaxws-api:jar:2.3.1] of location MavenDependencyRoots = [GroupId = javax.xml.ws, ArtifactId = jaxws-api, Version = 2.3.1, ArtifactType = jar, IncludeDependencyScope = ], IncludeDependencyScope = null, MissingManifestStrategy = GENERATE, IncludeSource = true is not a valid jar file -> [Help 1]

the real problem is that there is an invalid manifest header, but the way we currently use the p2 api the error is hidden from the user. This will probably also require a change in p2 to get better error feedback in this case,

**Comment by laeubi** [comment]
Beside the error reporting, this works a bit different from how m2e handles the case, if the manifest is invalid, m2e currently works as if there is no OSGi manifest at all, but tycho fails...

**Comment by akurtakov** [comment]
Is there a reason to keep this one still? If there is p2 bug for the issue please point to it.

**Comment by laeubi** [comment]
Yes we should keep this as it is an inconcsitenc between tycho<->m2e
